ops/ccs-ops-misc/load_test/common/db.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+"""common.db
+
+Utility module for executing database queries
+"""
+import psycopg2
+
+
+LIMIT = 100000  # global limit on the number of records to return
+
+def _execute(uri, query):
+    """
+    Execute a PSQL select statement and return its results
+    """
+    print('Collecting test data...')
+    conn = psycopg2.connect(uri)
+    with conn:
+        with conn.cursor() as cursor:
+            cursor.execute(query)
+            results = cursor.fetchall()
+            print(f'Returned {len(results)} results from the database for the test.')
+    conn.close()
```

Comment:
Shouldn't `with` call `close automatically?

```
  with psycopg2.connect(uri) as conn:
      with conn.cursor() as cursor:
          cursor.execute(query)
          results = cursor.fetchall()
          print(f'Returned {len(results)} results from the database for the test.')
```

Reply:
One would think so, but unfortunately they way it was implemented it only completes the transaction (commit or rollback based on if it was successful), one has to manually close the connection.  https://www.psycopg.org/docs/connection.html#the-connection-class